File: app/src/main/java/com/zillennium/utswap/screens/setting/MyWallet/MyWalletActivity.java
package com.zillennium.utswap.screens.setting.MyWallet;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.annotation.SuppressLint;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.ImageView;
import android.widget.LinearLayout;

import com.zillennium.utswap.R;
import com.zillennium.utswap.screens.setting.MyWallet.dialog.ExportTransactionBottomSheetDialog;
import com.zillennium.utswap.screens.setting.MyWallet.dialog.FilterBottomSheetDialog;
import com.zillennium.utswap.screens.setting.depositScreen.DepositActivity;
import com.zillennium.utswap.screens.setting.transferScreen.TransferActivity;
import com.zillennium.utswap.screens.setting.withdrawalScreen.WithdrawalActivity;

import java.util.ArrayList;


public class MyWalletActivity extends AppCompatActivity {

    private ImageView imgBack, imgExport, imgFilter, imgDeposit, imgWithdrawal, imgMore, imgTransfer, imgDropDown;
    private RecyclerView recyclerView;
    private LinearLayout linearAllWallet;
    private int count = 2;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_wallet_transaction);

        initView();

        initAction();

        setAdapter();
    }

    public void initView(){
        imgBack = findViewById(R.id.img_back);
        imgExport = findViewById(R.id.imgExport);
        imgFilter = findViewById(R.id.imgFilter);
        imgDeposit = findViewById(R.id.img_deposit);
        imgTransfer = findViewById(R.id.img_transfer);
        imgWithdrawal = findViewById(R.id.img_withdrawal);
        recyclerView = findViewById(R.id.list_transaction);
        imgDropDown = findViewById(R.id.img_drop_down);
        linearAllWallet = findViewById(R.id.linear_all_wallet);
    }

    public void initAction(){
        imgBack.setOnClickListener(view -> {finish();});

        imgFilter.setOnClickListener(view -> {
            FilterBottomSheetDialog a = FilterBottomSheetDialog.newInstance();
            a.show(getSupportFragmentManager(),"Open_fragment");
        });

        imgExport.setOnClickListener(view -> {
            ExportTransactionBottomSheetDialog a = ExportTransactionBottomSheetDialog.newInstance();
            a.show(getSupportFragmentManager(),"Open_fragment");
        });

        imgWithdrawal.setOnClickListener(view -> {
            Intent intent = new Intent(this, WithdrawalActivity.class);
            startActivity(intent);
        });

        imgTransfer.setOnClickListener(view -> {
            Intent intent = new Intent(this, TransferActivity.class);
            startActivity(intent);
        });

        imgDeposit.setOnClickListener(view -> {
            Intent intent = new Intent(this, DepositActivity.class);
            startActivity(intent);
        });

        imgDropDown.setOnClickListener(view -> {
            openDropDown(count);
            count++;
        });
    }

    @SuppressLint("UseCompatLoadingForDrawables")
    public void openDropDown(int count){
        if(count % 2 == 0)
        {
            linearAllWallet.setVisibility(View.GONE);
            imgDropDown.setImageDrawable(getDrawable(R.drawable.ic_baseline_keyboard_arrow_down_24));
        }
        else
        {
            linearAllWallet.setVisibility(View.VISIBLE);
            imgDropDown.setImageDrawable(getDrawable(R.drawable.ic_baseline_keyboard_arrow_up_24));
        }


    }

    public void setAdapter(){

        ArrayList<String> arrayDay = new ArrayList<>();

        String dayCurrent = "";
        ArrayList<MyWalletGroup> dataGroupList = new ArrayList<MyWalletGroup>();
        ArrayList<MyWalletItem> dataItemList = new ArrayList<MyWalletItem>();

        ArrayList<MyWalletItem> dataBalance = DataMyWallet.LIST_OF_BALANCE_HISTORY();

        for (int i = 0; i < dataBalance.size(); i++){
            MyWalletItem data = dataBalance.get(i);

            if(i == 0 || i == dataBalance.size() || dayCurrent.equals("") || data.getDate() != dayCurrent){
                if(i != 0){
                    dataGroupList.add(new MyWalletGroup(dayCurrent, dataItemList));
                    dataItemList = new ArrayList<>();
                }
            }
            dayCurrent = data.getDate();
            dataItemList.add(data);
        }

        LinearLayoutManager linearLayoutManager = new LinearLayoutManager(this);
        recyclerView.setLayoutManager(linearLayoutManager);
        MyWalletGroupAdapter myWalletGroupAdapter = new MyWalletGroupAdapter(dataGroupList);
        recyclerView.setAdapter(myWalletGroupAdapter);
    }

}

File: app/src/main/java/com/zillennium/utswap/screens/wallet/MyWallet/MyWalletActivity.kt
package com.zillennium.utswap.screens.wallet.MyWallet

import android.annotation.SuppressLint
import android.content.Intent
import android.view.View
import androidx.recyclerview.widget.LinearLayoutManager
import com.zillennium.utswap.R
import com.zillennium.utswap.UTSwapApp
import com.zillennium.utswap.bases.mvp.BaseMvpActivity
import com.zillennium.utswap.databinding.ActivityWalletTransactionBinding
import com.zillennium.utswap.screens.wallet.MyWallet.adapter.MyWalletGroupAdapter
import com.zillennium.utswap.screens.wallet.MyWallet.dialog.ExportTransactionBottomSheetDialog
import com.zillennium.utswap.screens.wallet.MyWallet.dialog.FilterBottomSheetDialog
import com.zillennium.utswap.screens.wallet.MyWallet.model.MyWalletGroup
import com.zillennium.utswap.screens.wallet.MyWallet.model.MyWalletItem
import com.zillennium.utswap.screens.wallet.depositScreen.DepositActivity
import com.zillennium.utswap.screens.wallet.transferScreen.TransferActivity
import com.zillennium.utswap.screens.wallet.withdrawalScreen.WithdrawalActivity
import java.io.IOException

class MyWalletActivity :
    BaseMvpActivity<MyWalletView.View, MyWalletView.Presenter, ActivityWalletTransactionBinding>(),
    MyWalletView.View {

    override var mPresenter: MyWalletView.Presenter = MyWalletPresenter()
    override val layoutResource: Int = R.layout.activity_wallet_transaction

    private var count = 2

    override fun initView() {
        super.initView()
        try {
            binding.apply {

                imgBack.setOnClickListener { finish() }

                imgFilter.setOnClickListener {
                    val a = FilterBottomSheetDialog.newInstance()
                    a.show(supportFragmentManager, "Open_fragment")
                }

                imgExport.setOnClickListener {
                    val a =
                        ExportTransactionBottomSheetDialog.newInstance()
                    a.show(supportFragmentManager, "Open_fragment")
                }

                imgWithdrawal.setOnClickListener {
                    val intent = Intent(UTSwapApp.instance, WithdrawalActivity::class.java)
                    startActivity(intent)
                }

                imgTransfer.setOnClickListener {
                    val intent = Intent(UTSwapApp.instance, TransferActivity::class.java)
                    startActivity(intent)
                }

                imgDeposit.setOnClickListener {
                    val intent = Intent(UTSwapApp.instance, DepositActivity::class.java)
                    startActivity(intent)
                }

                imgDropDown.setOnClickListener { view: View? ->
                    openDropDown(count)
                    count++
                }

            }
            // Code
        } catch (error: IOException) {
            // Must be safe
        }
    }

    @SuppressLint("UseCompatLoadingForDrawables")
    fun openDropDown(count: Int) {
        binding.apply {
            if (count % 2 == 0) {
                linearAllWallet.visibility = View.GONE
                imgDropDown.setImageDrawable(getDrawable(R.drawable.ic_baseline_keyboard_arrow_down_24))
            } else {
                linearAllWallet.visibility = View.VISIBLE
                imgDropDown.setImageDrawable(getDrawable(R.drawable.ic_baseline_keyboard_arrow_up_24))
            }
        }
    }

    fun setAdapter() {
        binding.apply {
            val arrayDay = ArrayList<String>()
            var dayCurrent = ""
            val dataGroupList = ArrayList<MyWalletGroup>()
            var dataItemList = ArrayList<MyWalletItem?>()
            val dataBalance = DataMyWallet.LIST_OF_BALANCE_HISTORY()
            for (i in dataBalance.indices) {
                val data = dataBalance[i]
                if (i == 0 || i == dataBalance.size || dayCurrent == "" || data.date !== dayCurrent) {
                    if (i != 0) {
                        dataGroupList.add(MyWalletGroup(dayCurrent, dataItemList))
                        dataItemList = ArrayList()
                    }
                }
                dayCurrent = data.date
                dataItemList.add(data)
            }
            val linearLayoutManager = LinearLayoutManager(UTSwapApp.instance)
            listTransaction.layoutManager = linearLayoutManager
            val myWalletGroupAdapter = MyWalletGroupAdapter(dataGroupList)
            listTransaction.adapter = myWalletGroupAdapter
        }

    }
}